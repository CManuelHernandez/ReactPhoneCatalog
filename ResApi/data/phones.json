{
  "mobilePhones": [
    {
      "name": "iPhone 13",
      "manufacturer": "apple",
      "description": "lorem ipsum dolor sit amet consectetur",
      "color": "black",
      "price": 845,
      "imageFileName": "images/iphone13.png",
      "screen": 6,
      "processor": "Apple A15",
      "ram": 4
    },
    {
      "name": "iPhone X",
      "manufacturer": "apple",
      "description": "lorem ipsum dolor sit amet consectetur",
      "color": "white",
      "price": 439,
      "imageFileName": "images/iphonex.jpg",
      "screen": 5,
      "processor": "Apple A11",
      "ram": 3
    },
    {
      "name": "Huawei p50",
      "manufacturer": "Huawei",
      "description": "lorem ipsum dolor sit amet consectetur",
      "color": "black",
      "price": 1100,
      "imageFileName": "images/hdp50.jpg",
      "screen": 7,
      "processor": "Snapdragon 888 4G",
      "ram": 8
    },
    {
      "name": "Xiaomi Poco X3",
      "manufacturer": "Xiaomi",
      "description": "lorem ipsum dolor sit amet consectetur",
      "color": "Blue",
      "price": 348,
      "imageFileName": "images/xiaomipocox3",
      "screen": 6,
      "processor": "Snapdragon 888 4G",
      "ram": 6
    }
  ]
}
