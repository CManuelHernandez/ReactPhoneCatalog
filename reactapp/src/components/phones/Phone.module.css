.phoneSection {
  position: relative;
  padding-top: 150px;
  padding-bottom: 30px;
}
.phoneCard {
  color: white;
  padding-top: 50px;
  padding-bottom: 50px;
  padding-left: 25px;
  padding-right: 25px;
  height: auto;
}
.phoneCardTitle {
  font-size: 2.2rem;
  font-weight: bolder;
}
.phoneCardView,
.phoneCardViewDetail {
  font-size: 1.1rem;
  font-weight: bold;
  box-shadow: 0 0 15px #1e780e;
  background-color: transparent;
  opacity: 0.9;
  transition: all 0.5s ease 0s;
  height: 100%;
  background-color: #b8bab7;
}
.phoneCardView:hover,
.phoneCardViewDetail:hover {
  transform: scale(1.02);
  overflow: hidden;
  box-shadow: 0 0 5px #1e780e;
}
.phoneCardView {
  background-color: #52564e;
}
.phoneImg {
  width: 100%;
  border-radius: 15px;
}
.phoneInfo {
  padding: 20px;
  height: auto;
}
.btnPrimary {
  display: flex;
  justify-content: center;
  margin-top: 0.75rem;
  color: black;
  font-weight: bold;
  background-color: #3be71d;
  border-color: #3be71d;
}
.btnDetail {
  position: absolute;
  right: 0;
  bottom: 0;
  margin: 0 0.75rem 0.75rem 0;
}
.btnPrimary:hover {
  color: black;
  background-color: #8fdc82;
  border-color: #8fdc82;
}
.btn:focus {
  outline: none;
  box-shadow: none;
}
.phoneHeading,
.phoneSubtitle {
  color: whitesmoke;
  font-size: 2.3em;
  font-weight: 500;
  padding-top: 10px;
}
.phoneSubtitle {
  text-align: center;
  font-size: 1.8em;
}
.phoneInfoTitle h1,
.phoneInfoTitle p {
  display: flex;
  justify-content: center;
}
.phoneDetailInfoText {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  height: 70%;
}
