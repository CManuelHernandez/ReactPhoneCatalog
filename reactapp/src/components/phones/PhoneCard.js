import React from "react";
import { Row, Col, Card, Button } from "react-bootstrap";
import classes from "./Phone.module.css";

export const PhoneCard = ({
  id,
  name,
  manufacturer,
  description,
  color,
  price,
  imageFileName,
  screen,
  processor,
}) => {
  return (
    <Card className={classes.phoneCardView}>
      <Card.Body>
        <Card.Title className={classes.phoneCardTitle}>{name}</Card.Title>
        <Row>
          <Col md={6} className={classes.phoneInfo}>
            <Card.Text>Manufacturer: {manufacturer}</Card.Text>
            <Card.Text>Color: {color}</Card.Text>
          </Col>
          <Col md={6} className={classes.phoneInfo}>
            <Card.Text>Screen: {screen} Inches</Card.Text>
            <Card.Text>Processor: {processor} Ram</Card.Text>
          </Col>
        </Row>
        <img
          className={classes.phoneImg}
          src={`http://localhost:3000/${imageFileName}`}
          alt={imageFileName}
        />
        <Card.Text style={{ textAlign: "center", marginTop: 10 }}>
          Description: {description}
        </Card.Text>
        <Card.Text style={{ textAlign: "right", fontSize: 40 }}>
          {price} â‚¬
        </Card.Text>
        <Button className={classes.btnPrimary} href={`/${id}`}>
          Detail
        </Button>
      </Card.Body>
    </Card>
  );
};
