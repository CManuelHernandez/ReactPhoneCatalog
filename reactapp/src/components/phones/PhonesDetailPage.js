import React from "react";
import { useParams, Navigate, useNavigate } from "react-router-dom";
import { Loading } from "../helpers/Loading";
import { Container, Row, Col, Card, Button } from "react-bootstrap";
import classes from "./Phone.module.css";

export const PhonesDetailPage = () => {
  const { id } = useParams();
  const navigate = useNavigate();

  const [phone, setPhone] = React.useState([]);
  const [loading, setLoading] = React.useState(false);

  React.useEffect(() => {
    //Api Call of the specific ID
    const getPhones = async () => {
      const url = `http://localhost:3000/phones/${id}`;
      const resp = await fetch(url);
      const results = await resp.json();
      setPhone(results.result);
    };
    getPhones();
    setLoading(true);
  }, [id]);

  const handleReturn = () => {
    navigate(-1);
  };

  if (!phone) {
    return <Navigate to="/" />;
  }

  return (
    <>
      <Container fluid className={classes.phoneSection}>
        <Container className={classes.phoneWrapper}>
          <Card className={classes.phoneCardViewDetail}>
            <Row>
              {loading ? (
                <>
                  <Col md={3} className={classes.phoneDetailImg}>
                    <Card.Img
                      className={classes.phoneImg}
                      src={`http://localhost:3000/${phone.imageFileName}`}
                      alt="card-img"
                    />
                  </Col>
                  <Col md={9} className={classes.phoneDetailInfo}>
                    <Card.Body className={classes.phoneDetailInfoText}>
                      <Card.Title className={classes.phoneCardTitle}>
                        {phone.name} from {phone.manufacturer}
                      </Card.Title>
                      <Card.Title>Color: {phone.color}</Card.Title>
                      <Card.Title>Screen: {phone.screen} Inches</Card.Title>
                      <Card.Title>Processor: {phone.processor} Ram</Card.Title>
                      <br />
                      <Card.Title>Description: {phone.description}</Card.Title>
                      <Card.Title style={{ textAlign: "right", fontSize: 50 }}>
                        {phone.price} â‚¬
                      </Card.Title>
                      <Button
                        className={`${classes.btnPrimary} ${classes.btnDetail}`}
                        onClick={handleReturn}
                      >
                        Retrun
                      </Button>
                    </Card.Body>
                  </Col>
                </>
              ) : (
                <Loading />
              )}
            </Row>
          </Card>
        </Container>
      </Container>
    </>
  );
};
