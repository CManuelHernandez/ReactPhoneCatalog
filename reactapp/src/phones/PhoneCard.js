import React from "react";
import Card from "react-bootstrap/Card";
import Button from "react-bootstrap/Button";
import classes from "./Phone.module.css";

export const PhoneCard = ({
  id,
  name,
  manufacturer,
  description,
  color,
  price,
  imageFileName,
  screen,
  processor,
}) => {
  return (
    <Card className={classes.phoneCardView}>
      <Card.Body>
        <Card.Title>
          <strong>NAME:</strong> {name}
        </Card.Title>
        <Card.Text>Manufacturer: {manufacturer}</Card.Text>
        <Card.Text>Color: {color}</Card.Text>
        <Card.Text>Screen: {screen} Inches</Card.Text>
        <Card.Text>EMAIL: {processor} Ram</Card.Text>
        <img
          className={classes.phoneImg}
          src={`http://localhost:3000/${imageFileName}`}
          alt={imageFileName}
        />
        <Card.Text>Description: {description}</Card.Text>
        <Card.Text>Price: {price} â‚¬</Card.Text>
        <Button className={classes.btnPrimary} href={`/${id}`}>
          Detail
        </Button>
      </Card.Body>
    </Card>
  );
};
