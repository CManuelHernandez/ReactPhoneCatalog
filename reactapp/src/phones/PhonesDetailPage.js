import React from "react";
import { useParams, Navigate, useNavigate } from "react-router-dom";

export const PhonesDetailPage = () => {
  const { id } = useParams();
  const navigate = useNavigate();

  const [phone, setPhone] = React.useState([]);

  React.useEffect(() => {
    //Api Call of the specific ID
    const getPhones = async () => {
      const url = `http://localhost:3000/phones/${id}`;
      const resp = await fetch(url);
      const results = await resp.json();
      setPhone(results.result);
    };
    getPhones();
  }, [id]);

  const handleReturn = () => {
    navigate(-1);
  };

  if (!phone) {
    return <Navigate to="/" />;
  }

  return (
    <div>
      <h1>Detalle</h1>
      <h3>{phone.name}</h3>
      <h4>{phone.manufacturer}</h4>
      <h4>{phone.description}</h4>
      <h4>{phone.color}</h4>
      <h4>{phone.price}</h4>
      <img src={`http://localhost:3000/${phone.imageFileName}`} />
      <h4>{phone.imageFileName}</h4>
      <h4>{phone.screen}</h4>
      <h4>{phone.processor}</h4>
      <button className="btn btn-outline-info" onClick={handleReturn}>
        Back to Main Page
      </button>
    </div>
  );
};
