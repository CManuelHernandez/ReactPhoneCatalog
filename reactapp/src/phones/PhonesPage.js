import React from "react";

export const PhonesPage = () => {
  const [phone, setPhones] = React.useState([]);

  React.useEffect(() => {
    getUsers();
  }, []);

  const getUsers = async () => {
    const url = "http://localhost:3000/phones";
    const resp = await fetch(url);
    const results = await resp.json();
    console.log("Phones", results);
    console.log("Response Phones", results);

    const apiPhones = results.map((phone) => {
      return {
        id: phone.id,
        name: phone.name,
        manufacturer: phone.manufacturer,
        description: phone.description,
        color: phone.color,
        price: phone.price,
        imageFileName: phone.imageFileName,
        screen: phone.screen,
        processor: phone.processor,
      };
    });
    setPhones(apiPhones);
  };

  return (
    <>
      <h2>ApiCallTest</h2>
      <ol>
        {phone.map((phone) => (
          <li key={phone.id}>
            Nombre: {phone.name} Manufacturer: {phone.manufacturer} Description:
            {phone.description} Color: {phone.color} Price: {phone.price}
            ImageFileName: {phone.imageFileName} Screen: {phone.screen}{" "}
            Processor: {phone.processor}
          </li>
        ))}
      </ol>
    </>
  );
};
