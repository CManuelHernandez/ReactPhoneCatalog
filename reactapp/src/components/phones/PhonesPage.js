import React from "react";
import { PhoneCard } from "./PhoneCard";
import { Loading } from "../helpers/Loading";
import { Container, Row, Col } from "react-bootstrap";

import classes from "./Phone.module.css";

export const PhonesPage = () => {
  const [phone, setPhones] = React.useState([]);
  const [loading, setLoading] = React.useState(false);

  React.useEffect(() => {
    getPhones();
  }, []);

  const getPhones = async () => {
    const url = "http://localhost:3000/phones";
    const resp = await fetch(url);
    const results = await resp.json();

    const apiPhones = results.map((phone) => {
      return {
        id: phone._id,
        name: phone.name,
        manufacturer: phone.manufacturer,
        description: phone.description,
        color: phone.color,
        price: phone.price,
        imageFileName: phone.imageFileName,
        screen: phone.screen,
        processor: phone.processor,
      };
    });
    setPhones(apiPhones);
    setLoading(true);
  };

  return (
    <>
      <Container fluid className={classes.phoneSection}>
        <Container className={classes.phoneWrapper}>
          <Col md={12} className={classes.phoneInfoTitle}>
            <div>
              <h1 className={classes.phoneHeading}>
                <strong className="turquoise"> REACT </strong> - Phone Catalog
              </h1>
              <h2 className={classes.phoneSubtitle}>
                Carlos Manuel Hern√°ndez Montero
              </h2>
            </div>
          </Col>

          <Row style={{ justifyContent: "center", paddingBottom: "10px" }}>
            {loading ? (
              phone.map((phone) => (
                <Col md={5} className={classes.phoneCard} key={phone.id}>
                  <PhoneCard
                    md={4}
                    className={classes.phoneCard}
                    name={phone.name}
                    manufacturer={phone.manufacturer}
                    description={phone.description}
                    color={phone.color}
                    price={phone.price}
                    imageFileName={phone.imageFileName}
                    screen={phone.screen}
                    processor={phone.processor}
                    {...phone}
                  />
                </Col>
              ))
            ) : (
              <Loading />
            )}
          </Row>
        </Container>
      </Container>
    </>
  );
};
