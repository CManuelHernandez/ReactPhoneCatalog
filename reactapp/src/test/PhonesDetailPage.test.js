import React from "react";
import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";

import { PhonesDetailPage } from "../components/phones/PhonesDetailPage";

const MockPhonesDetailPage = () => {
  return (
    <BrowserRouter>
      <PhonesDetailPage />
    </BrowserRouter>
  );
};

describe("PhonesDetailPage is render", () => {
  test("should return true", () => {
    render(<MockPhonesDetailPage />);
    const title = screen.getByText("Phone Detail");
    expect(title).toBeInTheDocument();
  });
});

describe("PhonesDetailPage show data from the async call", () => {
  test("should return true", async () => {
    render(<MockPhonesDetailPage />);
    const phonesPageDivElement = await screen.findByTestId("test");
    expect(phonesPageDivElement).toBeInTheDocument();
  });
});

describe("Test on <PhonesDetailPage />", () => {
  test("snapshot testing", () => {
    const tree = renderer.create(<MockPhonesDetailPage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
