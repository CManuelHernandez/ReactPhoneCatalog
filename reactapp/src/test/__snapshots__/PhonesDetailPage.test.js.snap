// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <PhonesDetailPage /> snapshot testing 1`] = `
<div
  className="phoneSection container-fluid"
>
  <h2
    className="phoneSubtitle"
  >
    Phone Detail
  </h2>
  <div
    className="phoneWrapper container"
  >
    <div
      className="phoneCardViewDetail card"
    >
      <div
        className="row"
      >
        <button
          className="btn btn-danger"
          disabled={true}
          type="button"
        >
          <span
            aria-hidden="true"
            className="spinner-grow spinner-grow-sm"
            role="status"
          />
          Loading...
        </button>
      </div>
    </div>
  </div>
</div>
`;
